src/database/schema/ChannelListSchema.ts

```diff
@@ -282,15 +370,18 @@ class ChannelList implements BaseDbSchema {
     });
   }
 
-  static fromPostNotifObject(json): ChannelList {
+  static fromPostNotifObject(
+    json,
+    channelType: 'POST_NOTIFICATION' | 'ANON_POST_NOTIFICATION'
+  ): ChannelList {
     const object = json?.new[0];
     return new ChannelList({
-      id: object?.id,
+      id: `${object?.id}${channelType === 'ANON_POST_NOTIFICATION' ? '_anon' : ''}'}`,
       channelPicture: '',
       name: '',
       description: object?.message,
       unreadCount: 1,
-      channelType: 'ANON_POST_NOTIFICATION',
+      channelType,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `fromPostNotifObject` function now accepts a `channelType` parameter which is a good improvement for flexibility and reusability of the function. The function implementation looks correct.



<!-- This is an auto-generated comment by OSS CodeRabbit -->